### PR TITLE
xds: implement STS based OAuth 2.0 credentials exchange

### PR DESCRIPTION
**Pull request by sanjaypujare**

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/internal/sts/StsCredentials.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      response = request.execute();
+    } catch (IOException e) {
+      throw new IOException("Error requesting access token", e);
+    }
+
+    if (response.getStatusCode() != HttpStatusCodes.STATUS_CODE_OK) {
+      throw new IOException("Error getting access token " + getStatusString(response));
+    }
+
+    GenericData responseData = response.parseAs(GenericData.class);
```
Comment (dapengzhang0) [inline xds/src/main/java/io/grpc/xds/internal/sts/StsCredentials.java:130]: Surround this line with `try{}` and next line with `finally{}`?

Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/internal/sts/StsCredentials.java:130]: This pattern was copied from other similar implementations in google-auth e.g. https://github.com/googleapis/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java#L264 

Many other implementations don't even disconnect! Anyway I'll use try-finally as suggested!

File: xds/src/main/java/io/grpc/xds/internal/sts/StsCredentials.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+// TODO(sanjaypujare): replace with the official implementation from google-auth once ready
+/** Implementation of OAuth2 Token Exchange as per https://tools.ietf.org/html/rfc8693. */
+public class StsCredentials extends GoogleCredentials {
```
[the remark this answers is not in the document]
Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/internal/sts/StsCredentials.java]: done

File: xds/src/main/java/io/grpc/xds/internal/sts/StsCredentials.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    HttpResponse response = null;
+    try {
+      response = request.execute();
+    } catch (IOException e) {
+      throw new IOException("Error requesting access token", e);
+    }
+
+    if (response.getStatusCode() != HttpStatusCodes.STATUS_CODE_OK) {
+      throw new IOException("Error getting access token " + getStatusString(response));
```
Comment (dapengzhang0) [inline xds/src/main/java/io/grpc/xds/internal/sts/StsCredentials.java:126]: Maybe add a colon "Error getting access token: " + getStatusString(response)

**Pull request merged**